diff --git a/node_modules/@subql/cli/lib/template/model.ts.ejs b/node_modules/@subql/cli/lib/template/model.ts.ejs
index 89e86fb..2254ab4 100644
--- a/node_modules/@subql/cli/lib/template/model.ts.ejs
+++ b/node_modules/@subql/cli/lib/template/model.ts.ejs
@@ -19,7 +19,7 @@ export class <%= props.className %> implements Entity {
 
     constructor(
         <% props.fields.forEach(function(field) { if (field.required) { %>
-            <%= field.name %>: <%= field.type %>,<% } %>
+            <%= field.name %>: <%= field.type %><%if (field.isArray) {%>[]<%}%>,<% } %>
         <% }) %>
     ) {
         <% props.fields.filter(function(field) {return field.required === true;}).forEach(function(requiredField) { %>
